File: scripts/extract_pdf.mjs
/** Extract text from HIT Complex Analysis PDF textbook. */
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const pdfPath = path.join(__dirname, '..', 'HIT教材 复变函数与积分变换同步学习(2).pdf');

async function main() {
  const pdfParse = (await import('pdf-parse')).default;
  const dataBuffer = fs.readFileSync(pdfPath);

  console.log(`Reading: ${pdfPath}`);
  console.log(`File size: ${(dataBuffer.length / 1024 / 1024).toFixed(1)} MB`);

  const data = await pdfParse(dataBuffer);
  console.log(`Total pages: ${data.numpages}`);
  console.log(`Total chars: ${data.text.length}`);
  console.log(`\n=== Info ===`);
  console.log(JSON.stringify(data.info, null, 2));
  console.log(`\n=== First 5000 chars ===`);
  console.log(data.text.slice(0, 5000));

  // Save full text to file
  const outPath = path.join(__dirname, '..', 'data', 'hit_textbook_full.txt');
  const outDir = path.dirname(outPath);
  if (!fs.existsSync(outDir)) fs.mkdirSync(outDir, { recursive: true });
  fs.writeFileSync(outPath, data.text, 'utf-8');
  console.log(`\nFull text saved to: ${outPath} (${(data.text.length / 1024).toFixed(0)} KB)`);
}

main().catch(err => { console.error(err); process.exit(1); });

File: scripts/extract_pdf.py
"""Extract text from HIT Complex Analysis PDF textbook."""
import sys
import os

# Add backend to path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'backend'))

try:
    import PyPDF2
except ImportError:
    print("PyPDF2 not found, trying pdfplumber...")
    try:
        import pdfplumber
        HAS_PDFPLUMBER = True
        HAS_PYPDF = False
    except ImportError:
        print("No PDF library available. Install PyPDF2 or pdfplumber.")
        sys.exit(1)
else:
    HAS_PYPDF = True
    HAS_PDFPLUMBER = False

PDF_PATH = os.path.join(os.path.dirname(__file__), '..', 'HIT教材 复变函数与积分变换同步学习(2).pdf')

def extract_with_pypdf(path):
    reader = PyPDF2.PdfReader(path)
    print(f"Total pages: {len(reader.pages)}")
    for i in range(min(10, len(reader.pages))):
        text = reader.pages[i].extract_text()
        print(f"\n{'='*60}")
        print(f"PAGE {i+1}")
        print(f"{'='*60}")
        if text:
            print(text[:2000])
        else:
            print("[No text extracted]")

def extract_with_pdfplumber(path):
    with pdfplumber.open(path) as pdf:
        print(f"Total pages: {len(pdf.pages)}")
        for i in range(min(10, len(pdf.pages))):
            text = pdf.pages[i].extract_text()
            print(f"\n{'='*60}")
            print(f"PAGE {i+1}")
            print(f"{'='*60}")
            if text:
                print(text[:2000])
            else:
                print("[No text extracted]")

if __name__ == '__main__':
    if HAS_PYPDF:
        extract_with_pypdf(PDF_PATH)
    else:
        extract_with_pdfplumber(PDF_PATH)
